zombies.js

```diff
@@ -79,135 +99,231 @@
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice
+   *
+   * If the item is not in the pack, return a message with the item name saying
+   *   nothing was discarded since the item could not be found.
+   * Return false in this case.
+   *
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+
+  discardItem(item) {
+    let index = this.getPack().indexOf(item)
+    if (index === -1) {
+      console.log('item not here');
+      return false;
+    } else {
+      this.getPack().splice(index, 1);
+      console.log(this.name + 'discarded' + item)
+      return true;
+    }
+  }
+
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  equip(itemToEquip) {
+    let index = this.getPack().indexOf(itemToEquip)
+    let spliced = this.getPack().splice(index, 1, this.equipped)[0]
+    if (itemToEquip instanceof(Weapon) && index !== -1) {
+      // if itemtoequip is a weapon and it's in my pack
+      if (this.equipped !== false) {
+        this.equipped = spliced;
+        // if item is equipped with another weapon, replace it with your itemtoequip
+      } else {
+        // 
+        this.equipped = itemToEquip
+      }
+    }
+  }
+
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+
+  eat(itemToEat) {
+    let index = this.getPack().indexOf(itemToEat)
```

Comment:
This is unnecessary since the .includes in the line below essentially does the same thing.